README.md
# eGHL Flutter Plugin

An example of Flutter integration with eGHL SDK

## Guidelines linking to SDK

Refer [here](https://github.com/eGHL/flutter-plugin-eghl/wiki/Android-&-iOS-SDK-library-linking) on how to link for iOS & Android platform

## Usage 
---------------

### Request Payment

```
String paymentId = 'DEMO${new DateTime.now().millisecondsSinceEpoch}';

Map<String, dynamic> payment = {
  'TransactionType': 'SALE',
  'Amount': '1.00',
  'CurrencyCode': 'MYR',
  'PaymentId': paymentId,
  'OrderNumber': paymentId,
  'PaymentDesc': 'Testing Payment',
  'PymtMethod': 'ANY',

  'CustName': 'somebody',
  'CustEmail': 'somebody@someone.com',
  'CustPhone': '0123456789',
  'MerchantReturnURL': 'SDK',   //merchant system's browser redirect
  'MerchantCallBackURL': 'SDK', //server-to-server URL as an additional link to merchant's website

  'ServiceId': 'SIT',
  'Password': 'sit12345',
  'LanguageCode': 'EN',
  'PageTimeout': '600',
  'PaymentGateway': true,      //default is true for staging, false for production
  'EnableCardPage': false,    //default is false
  'TriggerReturnURL': false, //default is false
};

result = await Eghlflutter.executePayment(payment);


CHANGELOG.md
## 1.0.0

* Initial commit of eGHL Flutter Plugin

## 1.0.1

* Include EnableCardPage param
* Change PaymentGateway value to boolean

## 1.0.2

* Removing redundant class in iOS

## 1.0.3

* Include CardPage for Android and iOS

## 1.0.4

* Include EnabledCardPage params for Android

## 1.0.5

* Fix onActivityResult to return status when Payment Cancelled

## 1.0.6

* Removed folder from .gitignore 

ios/Classes/EghlflutterPlugin.m
#import "EghlflutterPlugin.h"
#import <objc/runtime.h>
#import <EGHL/EGHL.h>

 @implementation EghlflutterPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"eghlflutter"
            binaryMessenger:[registrar messenger]];
  EghlflutterPlugin* instance = [[EghlflutterPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"executePayment" isEqualToString:call.method]) {
      PaymentRequestPARAM *payParams = [[PaymentRequestPARAM alloc] init];
      EGHLPayment *eghlPay = [[EGHLPayment alloc] init];
      
      payParams.TransactionType = [call.arguments valueForKey:@"TransactionType"];
      payParams.PaymentID = [call.arguments valueForKey:@"PaymentId"];
      payParams.OrderNumber = [call.arguments valueForKey:@"OrderNumber"];
      payParams.ServiceID = [call.arguments valueForKey:@"ServiceId"];
      payParams.MerchantName = [call.arguments valueForKey:@"MerchantName"];
      payParams.PymtMethod = [call.arguments valueForKey:@"PymtMethod"];
      payParams.MerchantReturnURL = [call.arguments valueForKey:@"MerchantReturnURL"];
      payParams.MerchantCallBackURL = [call.arguments valueForKey:@"MerchantCallBackURL"];
      payParams.MerchantApprovalURL = [call.arguments valueForKey:@"MerchantApprovalURL"];
      payParams.MerchantUnApprovalURL = [call.arguments valueForKey:@"MerchantUnApprovalURL"];
      payParams.CustMAC = [call.arguments valueForKey:@"CustMAC"];
      payParams.PageTimeout = [call.arguments valueForKey:@"PageTimeout"];
      payParams.CustEmail = [call.arguments valueForKey:@"CustEmail"];
      payParams.CustPhone = [call.arguments valueForKey:@"CustPhone"];
      payParams.CustIP = [call.arguments valueForKey:@"CustIP"];
      payParams.CustName = [call.arguments valueForKey:@"CustName"];
      payParams.PaymentDesc = [call.arguments valueForKey:@"PaymentDesc"];
      payParams.Amount = [call.arguments valueForKey:@"Amount"];
      payParams.EPPMonth = [call.arguments valueForKey:@"EPPMonth"];
      payParams.CurrencyCode = [call.arguments valueForKey:@"CurrencyCode"];
      payParams.LanguageCode = [call.arguments valueForKey:@"LanguageCode"];
      payParams.IssuingBank = [call.arguments valueForKey:@"IssuingBank"];
      payParams.Password = [call.arguments valueForKey:@"Password"];
      payParams.CardHolder = [call.arguments valueForKey:@"CardHolder"];
      payParams.CardNo = [call.arguments valueForKey:@"CardNo"];
      payParams.CardExp = [call.arguments valueForKey:@"CardExp"];
      payParams.CardCVV2 = [call.arguments valueForKey:@"CardCVV2"];
      payParams.BillAddr = [call.arguments valueForKey:@"BillAddr"];
      payParams.BillPostal = [call.arguments valueForKey:@"BillPostal"];
      payParams.BillCity = [call.arguments valueForKey:@"BillCity"];
      payParams.BillRegion = [call.arguments valueForKey:@"BillRegion"];
      payParams.BillCountry = [call.arguments valueForKey:@"BillCountry"];
      payParams.PromoCode = [call.arguments valueForKey:@"PromoCode"];
      payParams.ShipAddr = [call.arguments valueForKey:@"ShipAddr"];
      payParams.ShipPostal = [call.arguments valueForKey:@"ShipPostal"];
      payParams.ShipCity = [call.arguments valueForKey:@"ShipCity"];
      payParams.ShipRegion = [call.arguments valueForKey:@"ShipRegion"];
      payParams.ShipCountry = [call.arguments valueForKey:@"ShipCountry"];
      payParams.Token = [call.arguments valueForKey:@"Token"];
      payParams.TokenType = [call.arguments valueForKey:@"TokenType"];
      payParams.SessionID = [call.arguments valueForKey:@"SessionID"];
      payParams.B4TaxAmt = [call.arguments valueForKey:@"B4TaxAmt"];
      payParams.Param6 = [call.arguments valueForKey:@"Param6"];
      payParams.Param7 = [call.arguments valueForKey:@"Param7"];
      payParams.ReqToken = [call.arguments valueForKey:@"ReqToken"];
      payParams.ReqVerifier = [call.arguments valueForKey:@"ReqVerifier"];
      payParams.PairingToken = [call.arguments valueForKey:@"PairingToken"];
      payParams.PairingVerifier = [call.arguments valueForKey:@"PairingVerifier"];
      payParams.CheckoutResourceURL = [call.arguments valueForKey:@"CheckoutResourceURL"];
      payParams.CardId = [call.arguments valueForKey:@"CardId"];
      payParams.PreCheckoutId = [call.arguments valueForKey:@"PreCheckoutId"];
      payParams.mpLightboxParameter = [call.arguments valueForKey:@"mpLightboxParameter"];
      payParams.sdkTimeOut = [[call.arguments objectForKey:@"sdkTimeout"] doubleValue];
      BOOL gatewayUrl = [[call.arguments valueForKey:@"PaymentGateway"] boolValue];
      BOOL cardPageEnable = [[call.arguments valueForKey:@"EnableCardPage"] boolValue];
      BOOL triggerReturnUrl = [[call.arguments valueForKey:@"TriggerReturnURL"] boolValue];
      payParams.settingDict = @{
          EGHL_DEBUG_PAYMENT_URL: [NSNumber numberWithBool:gatewayUrl],
          EGHL_ENABLED_CARD_PAGE: [NSNumber numberWithBool:cardPageEnable],
          EGHL_SHOULD_TRIGGER_RETURN_URL: [NSNumber numberWithBool:triggerReturnUrl],
      };

      
      UIWindow *keyWindow = [[UIApplication sharedApplication] keyWindow];

      UIViewController *vc = keyWindow.rootViewController;
      
      [eghlPay execute:payParams fromViewController:vc successBlock:^(PaymentRespPARAM *responseData) {
          //Convert to NSData
          NSMutableDictionary *dict = [NSMutableDictionary dictionary];
          unsigned count;
          objc_property_t *properties = class_copyPropertyList([responseData class], &count);
          for(int i=0; i<count; i++) {
              NSString* key = [NSString stringWithUTF8String:property_getName(properties[i])];
              NSString *value = [responseData valueForKey:key];
              
              if(value)[dict setObject:value forKey:key];
          }
          
          NSError* error;
          NSData* responseJsonData = [NSJSONSerialization dataWithJSONObject:dict options:kNilOptions error:&error];
          
          result([[NSString alloc] initWithData:responseJsonData encoding:NSUTF8StringEncoding]);
          
      }failedBlock:^(NSString *errorCode, NSString *errorData, NSError *error) {
          result([FlutterError errorWithCode:[NSString stringWithFormat:@"Error %ld", error.code]
          message:error.domain details: nil]);
      }];
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end


android/src/main/java/com/ghl/eghl/eghlflutter/EghlPaymentParam.java
package com.ghl.eghl.eghlflutter;

import com.google.gson.annotations.SerializedName;

public class EghlPaymentParam {

    /*Mandatory fields*/
    @SerializedName("TransactionType")
    private String transactionType = "";

    @SerializedName("PymtMethod")
    private String paymentMethod = "";

    @SerializedName("ServiceId")
    private String serviceId = "";

    @SerializedName("PaymentId")
    private String paymentId = "";

    @SerializedName("OrderNumber")
    private String orderNumber = "";

    @SerializedName("PaymentDesc")
    private String paymentDesc = "";

    @SerializedName("MerchantReturnURL")
    private String merchantReturnUrl = "";

    @SerializedName("Amount")
    private String amount = "";

    @SerializedName("CurrencyCode")
    private String currencyCode = "";

    @SerializedName("Password")
    private String password = "";

    @SerializedName("PaymentGateway")
    private boolean paymentGateway = true;

    @SerializedName("Production")
    private boolean isProduction = true;

    /*Optional Fields*/
    @SerializedName("CustIp")
    private String custIp = "";

    @SerializedName("CustName")
    private String custName = "";

    @SerializedName("CustEmail")
    private String custEmail = "";

    @SerializedName("CustPhone")
    private String custPhone = "";

    @SerializedName("B4TaxAmt")
    private String b4TaxAmt = "";

    @SerializedName("TaxAmt")
    private String taxAmt = "";

    @SerializedName("MerchantName")
    private String merchantName = "";

    @SerializedName("CustMAC")
    private String custMac = "";

    @SerializedName("MerchantApprovalURL")
    private String merchantApprovalUrl = "";

    @SerializedName("MerchantUnApprovalURL")
    private String merchantUnapprovalUrl = "";

    @SerializedName("MerchantCallBackURL")
    private String merchantCallbackUrl = "";

    @SerializedName("LanguageCode")
    private String languageCode = "";

    @SerializedName("PageTimeout")
    private String pageTimeout = "";

    @SerializedName("PaymentTimeout")
    private int paymentTimeout = -1;

    @SerializedName("EnableCardPage")
    private boolean cardPage = false;

    @SerializedName("CardHolder")
    private String cardHolder = "";

    @SerializedName("CardNo")
    private String cardNo = "";

    @SerializedName("CardExp")
    private String cardExp = "";

    @SerializedName("CardCVV2")
    private String cardCvv2 = "";

    @SerializedName("IssuingBank")
    private String issuingBank = "";

    @SerializedName("BillAddr")
    private String billAddr = "";

    @SerializedName("BillPostal")
    private String billPostal = "";

    @SerializedName("BillCity")
    private String billCity = "";

    @SerializedName("BillRegion")
    private String billRegion = "";

    @SerializedName("BillCountry")
    private String billCountry = "";

    @SerializedName("ShipAddr")
    private String shipAddr = "";

    @SerializedName("ShipPostal")
    private String shipPostal = "";

    @SerializedName("ShipCity")
    private String shipCity = "";

    @SerializedName("ShipRegion")
    private String shipRegion = "";

    @SerializedName("ShipCountry")
    private String shipCountry = "";

    @SerializedName("SessionID")
    private String sessionId = "";

    @SerializedName("TokenType")
    private String tokenType = "";

    /*Conditional fields*/
    @SerializedName("HashValue")
    private String hashValue = "";

    @SerializedName("Token")
    private String token = "";

    @SerializedName("Param6")
    private String param6 = "";

    @SerializedName("Param7")
    private String param7 = "";

    @SerializedName("EPPMonth")
    private String eppMonth = "";

    @SerializedName("PromoCode")
    private String promoCode = "";

    @SerializedName("TriggerReturnURL")
    private boolean triggerReturnURL = false;

    public boolean getProduction() {
        return isProduction;
    }

    public void setProduction(boolean production) {
        this.isProduction = production;
    }

    public boolean getPaymentGateway() {
        return paymentGateway;
    }

    public void setPaymentGateway(boolean paymentGateway) {
        this.paymentGateway = paymentGateway;
    }

    public String getCustIp() {
        return custIp;
    }

    public void setCustIp(String custIp) {
        this.custIp = custIp;
    }

    public String getCurrencyCode() {
        return currencyCode;
    }

    public void setCurrencyCode(String currencyCode) {
        this.currencyCode = currencyCode;
    }

    public String getPaymentMethod() {
        return paymentMethod;
    }

    public void setPaymentMethod(String paymentMethod) {
        this.paymentMethod = paymentMethod;
    }

    public String getTransactionType() {
        return transactionType;
    }

    public void setTransactionType(String transactionType) {
        this.transactionType = transactionType;
    }

    public String getServiceId() {
        return serviceId;
    }

    public void setServiceId(String serviceId) {
        this.serviceId = serviceId;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getPaymentId() {
        return paymentId;
    }

    public void setPaymentId(String paymentId) {
        this.paymentId = paymentId;
    }

    public String getOrderNumber() {
        return orderNumber;
    }

    public void setOrderNumber(String orderNumber) {
        this.orderNumber = orderNumber;
    }

    public String getPaymentDesc() {
        return paymentDesc;
    }

    public void setPaymentDesc(String paymentDesc) {
        this.paymentDesc = paymentDesc;
    }

    public String getMerchantReturnUrl() {
        return merchantReturnUrl;
    }

    public void setMerchantReturnUrl(String merchantReturnUrl) {
        this.merchantReturnUrl = merchantReturnUrl;
    }

    public String getAmount() {
        return amount;
    }

    public void setAmount(String amount) {
        this.amount = amount;
    }

    public String getCustName() {
        return custName;
    }

    public void setCustName(String custName) {
        this.custName = custName;
    }

    public String getCustEmail() {
        return custEmail;
    }

    public void setCustEmail(String custEmail) {
        this.custEmail = custEmail;
    }

    public String getCustPhone() {
        return custPhone;
    }

    public void setCustPhone(String custPhone) {
        this.custPhone = custPhone;
    }

    public String getB4TaxAmt() {
        return b4TaxAmt;
    }

    public void setB4TaxAmt(String b4TaxAmt) {
        this.b4TaxAmt = b4TaxAmt;
    }

    public String getTaxAmt() {
        return taxAmt;
    }

    public void setTaxAmt(String taxAmt) {
        this.taxAmt = taxAmt;
    }

    public String getMerchantName() {
        return merchantName;
    }

    public void setMerchantName(String merchantName) {
        this.merchantName = merchantName;
    }

    public String getCustMac() {
        return custMac;
    }

    public void setCustMac(String custMac) {
        this.custMac = custMac;
    }

    public String getMerchantApprovalUrl() {
        return merchantApprovalUrl;
    }

    public boolean getTriggerReturnUrl() {
        return triggerReturnURL;
    }

    public void setMerchantApprovalUrl(String merchantApprovalUrl) {
        this.merchantApprovalUrl = merchantApprovalUrl;
    }

    public String getMerchantUnapprovalUrl() {
        return merchantUnapprovalUrl;
    }

    public void setMerchantUnapprovalUrl(String merchantUnapprovalUrl) {
        this.merchantUnapprovalUrl = merchantUnapprovalUrl;
    }

    public String getMerchantCallbackUrl() {
        return merchantCallbackUrl;
    }

    public void setMerchantCallbackUrl(String merchantCallbackUrl) {
        this.merchantCallbackUrl = merchantCallbackUrl;
    }

    public String getLanguageCode() {
        return languageCode;
    }

    public void setLanguageCode(String languageCode) {
        this.languageCode = languageCode;
    }

    public String getPageTimeout() {
        return pageTimeout;
    }

    public void setPageTimeout(String pageTimeout) {
        this.pageTimeout = pageTimeout;
    }

    public void setCardPage(Boolean cardPage) {
        this.cardPage = cardPage;
    }

    public boolean getCardPage() { return cardPage; }

    public String getCardHolder() {
        return cardHolder;
    }

    public void setCardHolder(String cardHolder) {
        this.cardHolder = cardHolder;
    }

    public String getCardNo() {
        return cardNo;
    }

    public void setCardNo(String cardNo) {
        this.cardNo = cardNo;
    }

    public String getCardExp() {
        return cardExp;
    }

    public void setCardExp(String cardExp) {
        this.cardExp = cardExp;
    }

    public String getCardCvv2() {
        return cardCvv2;
    }

    public void setCardCvv2(String cardCvv2) {
        this.cardCvv2 = cardCvv2;
    }

    public String getIssuingBank() {
        return issuingBank;
    }

    public void setIssuingBank(String issuingBank) {
        this.issuingBank = issuingBank;
    }

    public String getBillAddr() {
        return billAddr;
    }

    public void setBillAddr(String billAddr) {
        this.billAddr = billAddr;
    }

    public String getBillPostal() {
        return billPostal;
    }

    public void setBillPostal(String billPostal) {
        this.billPostal = billPostal;
    }

    public String getBillCity() {
        return billCity;
    }

    public void setBillCity(String billCity) {
        this.billCity = billCity;
    }

    public String getBillRegion() {
        return billRegion;
    }

    public void setBillRegion(String billRegion) {
        this.billRegion = billRegion;
    }

    public String getBillCountry() {
        return billCountry;
    }

    public void setBillCountry(String billCountry) {
        this.billCountry = billCountry;
    }

    public String getHashValue() {
        return hashValue;
    }

    public void setHashValue(String hashValue) {
        this.hashValue = hashValue;
    }

    public String getShipAddr() {
        return shipAddr;
    }

    public void setShipAddr(String shipAddr) {
        this.shipAddr = shipAddr;
    }

    public String getShipPostal() {
        return shipPostal;
    }

    public void setShipPostal(String shipPostal) {
        this.shipPostal = shipPostal;
    }

    public String getShipCity() {
        return shipCity;
    }

    public void setShipCity(String shipCity) {
        this.shipCity = shipCity;
    }

    public String getShipRegion() {
        return shipRegion;
    }

    public void setShipRegion(String shipRegion) {
        this.shipRegion = shipRegion;
    }

    public String getShipCountry() {
        return shipCountry;
    }

    public void setShipCountry(String shipCountry) {
        this.shipCountry = shipCountry;
    }

    public String getSessionId() {
        return sessionId;
    }

    public void setSessionId(String sessionId) {
        this.sessionId = sessionId;
    }

    public String getTokenType() {
        return tokenType;
    }

    public void setTokenType(String tokenType) {
        this.tokenType = tokenType;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public String getParam6() {
        return param6;
    }

    public void setParam6(String param6) {
        this.param6 = param6;
    }

    public String getParam7() {
        return param7;
    }

    public void setParam7(String param7) {
        this.param7 = param7;
    }

    public String getEppMonth() {
        return eppMonth;
    }

    public void setEppMonth(String eppMonth) {
        this.eppMonth = eppMonth;
    }

    public String getPromoCode() {
        return promoCode;
    }

    public void setPromoCode(String promoCode) {
        this.promoCode = promoCode;
    }

    public int getPaymentTimeout() {
        return paymentTimeout;
    }

    public void setPaymentTimeout(int paymentTimeout) {
        this.paymentTimeout = paymentTimeout;
    }

}



android/src/main/java/com/ghl/eghl/eghlflutter/EghlflutterPlugin.java
package com.ghl.eghl.eghlflutter;

import android.app.Activity;
import android.content.Intent;
import android.text.TextUtils;
import android.util.Log;

import com.eghl.sdk.EGHL;
import com.eghl.sdk.params.PaymentParams;
import com.eghl.sdk.payment.PaymentActivity;
import com.google.gson.Gson;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/** EghlflutterPlugin */
public class EghlflutterPlugin implements MethodCallHandler, PluginRegistry.ActivityResultListener {
  private Activity _activity;
  private EghlPaymentParam paymentParam;
  private static final String TAG = "eGHLModule";
  private Result _result;


  private EghlflutterPlugin(Activity activity) {
    this._activity = activity;
  }

  public static void registerWith(Registrar registrar) {
    final MethodChannel channel = new MethodChannel(registrar.messenger(), "eghlflutter");
    EghlflutterPlugin plugin = new EghlflutterPlugin(registrar.activity());
    registrar.addActivityResultListener(plugin);
    channel.setMethodCallHandler(plugin);
  }

  @Override
  public void onMethodCall(MethodCall call, Result result) {
    if (call.method.equals("executePayment")) {
        this._result = result;
        paymentParam = new Gson().fromJson(new Gson().toJsonTree(call.arguments()).getAsJsonObject().toString(), EghlPaymentParam.class);

        try {
          PaymentParams.Builder params = new PaymentParams.Builder()
                  /*Mandatory fields*/
                  .setTransactionType(paymentParam.getTransactionType())
                  .setPaymentMethod(paymentParam.getPaymentMethod())
                  .setServiceId(paymentParam.getServiceId())
                  .setPaymentId(paymentParam.getPaymentId())
                  .setOrderNumber(paymentParam.getOrderNumber())
                  .setPaymentDesc(paymentParam.getPaymentDesc())
                  .setMerchantReturnUrl(paymentParam.getMerchantReturnUrl())
                  .setMerchantCallbackUrl(paymentParam.getMerchantCallbackUrl())
                  .setAmount(paymentParam.getAmount())
                  .setCurrencyCode(paymentParam.getCurrencyCode())
                  .setPassword(paymentParam.getPassword())
                  .setLanguageCode(paymentParam.getLanguageCode())
                  .setDebugPaymentURL(paymentParam.getPaymentGateway())
                  .setPageTimeout(paymentParam.getPageTimeout())
                  .setCardPageEnabled(paymentParam.getCardPage())

                  /*Optional fields*/
                  .setCustIp(paymentParam.getCustIp())
                  .setCustName(paymentParam.getCustName())
                  .setCustEmail(paymentParam.getCustEmail())
                  .setCustPhone(paymentParam.getCustPhone())
                  .setB4TaxAmt(paymentParam.getB4TaxAmt())
                  .setTaxAmt(paymentParam.getTaxAmt())
                  .setMerchantName(paymentParam.getMerchantName())
                  .setCustMac(paymentParam.getCustMac())
                  .setMerchantApprovalUrl(paymentParam.getMerchantApprovalUrl())
                  .setMerchantUnapprovalUrl(paymentParam.getMerchantUnapprovalUrl())
                  .setPaymentTimeout(paymentParam.getPaymentTimeout())
                  .setCardHolder(paymentParam.getCardHolder())
                  .setCardNo(paymentParam.getCardNo())
                  .setCardExp(paymentParam.getCardExp())
                  .setCardCvv2(paymentParam.getCardCvv2())
                  .setIssuingBank(paymentParam.getIssuingBank())
                  .setBillAddr(paymentParam.getBillAddr())
                  .setBillPostal(paymentParam.getBillPostal())
                  .setBillCity(paymentParam.getBillCity())
                  .setBillRegion(paymentParam.getBillRegion())
                  .setBillCountry(paymentParam.getBillCountry())
                  .setShipAddr(paymentParam.getShipAddr())
                  .setShipPostal(paymentParam.getShipPostal())
                  .setShipCity(paymentParam.getShipCity())
                  .setShipRegion(paymentParam.getShipRegion())
                  .setShipCountry(paymentParam.getShipCountry())
                  .setSessionId(paymentParam.getSessionId())
                  .setTokenType(paymentParam.getTokenType())
                  .setToken(paymentParam.getToken())
                  .setParam6(paymentParam.getParam6())
                  .setParam7(paymentParam.getParam7())
                  .setEppMonth(paymentParam.getEppMonth())

                  /*Conditional fields*/
                  .setPromoCode(paymentParam.getPromoCode())
                  .setTriggerReturnURL(paymentParam.getTriggerReturnUrl());

          Intent payment = new Intent(this._activity, PaymentActivity.class);
          payment.putExtras(params.build());

          this._activity.startActivityForResult(payment, EGHL.REQUEST_PAYMENT);
        } catch (Exception e) {
          this._result.error("eGHLModule", e.getMessage(), null);
        }
    }  else {
      result.notImplemented();
    }
  }

  @Override
  public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
    if(_result == null) {
      return false;
    }
    else if (requestCode == EGHL.REQUEST_PAYMENT && data != null) {
      String resultString;
      if (!TextUtils.isEmpty(data.getStringExtra(EGHL.RAW_RESPONSE))) {
        String message = data.getStringExtra(EGHL.TXN_MESSAGE);
        switch (resultCode) {
          case EGHL.TRANSACTION_SUCCESS:
            if(message == null) {
              Log.d(TAG, "onActivityResult: payment success");
              this._result.error("PAYMENT_Success", String.valueOf(resultCode), null);
            } else {
              Log.d(TAG, "onActivityResult: payment successful");
              resultString = data.getStringExtra(EGHL.RAW_RESPONSE);
              this._result.success(resultString);
            }
            break;
          case EGHL.TRANSACTION_FAILED:
            if(message == null) {
              Log.d(TAG, "onActivityResult: payment failure");
              this._result.error("PAYMENT_FAILED", String.valueOf(resultCode), null);
            } else {
              Log.d(TAG, "onActivityResult: payment failure or cancelled '"+message+"'");
              resultString = data.getStringExtra(EGHL.RAW_RESPONSE);
              this._result.success(resultString);
            }

            break;
          default:
            Log.d(TAG, "onActivityResult: " + resultCode);
            if(message == null) {
              this._result.error("UNKNOWN_FAILED", String.valueOf(resultCode), null);
            } else {
              resultString = data.getStringExtra(EGHL.RAW_RESPONSE);
              this._result.success(resultString);
            }
            break;
        }
      } else {
        switch (resultCode) {
          case EGHL.TRANSACTION_CANCELLED:
            String message = data.getStringExtra(EGHL.TXN_MESSAGE);
            this._result.success(message + resultCode);
            break;
        }
      }
    }
    this._result = null;
    return false;
  }
}



example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:eghlflutter/eghlflutter.dart';

void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  String _eghlPaymentResult = 'Awaiting for payment.';

  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text('eGHL Payment'),
        ),
        body: Container(
          child: Center (
            child: Column (
              mainAxisAlignment:  MainAxisAlignment.center,
              children: [
                Text(this._eghlPaymentResult),
                RaisedButton(
                  child: Text("Pay"),
                  onPressed: () {
                    _Pay(context);
                  },
                  color: Colors.amberAccent,
                  textColor: Colors.black,
                  padding: EdgeInsets.fromLTRB(10, 10, 10, 10),
                  splashColor: Colors.amber,
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }

  // ignore: non_constant_identifier_names
  void _Pay(BuildContext context) async {
    String result = '';
    String paymentId = 'SIT${new DateTime.now().millisecondsSinceEpoch}';

    try {
      Map<String, dynamic> payment = {
        'TransactionType': 'SALE',
        'Amount': '1.00',
        'CurrencyCode': 'MYR',
        'PaymentId': paymentId,
        'OrderNumber': paymentId,
        'PaymentDesc': 'Testing Payment',
        'PymtMethod': 'ANY',

        'CustName': 'somebody',
        'CustEmail': 'somebody@someone.com',
        'CustPhone': '0123456789',

        'MerchantReturnURL': 'SDK',
        'MerchantCallBackURL': 'SDK',

        'ServiceId': 'SIT',
        'Password': 'sit12345',
        'LanguageCode': 'EN',
        'PageTimeout': '600',
        'PaymentGateway': true,
        'EnableCardPage': false,
        'TriggerReturnURL' : false,
      };

      result = await Eghlflutter.executePayment(payment);
    } catch(e) {
      result = e.message;
    }

    setState(() {
      _eghlPaymentResult = result;
    });
  }
}
